python/ray/data/_internal/arrow_block.py

```diff
@@ -633,13 +621,13 @@ def aggregate_combined_blocks(
         """
 
         stats = BlockExecStats.builder()
+        keys = sort_key.get_columns()
 
-        keys = key if isinstance(key, list) else [key]
-        key_fn = (
-            (lambda r: tuple(r[r._row.schema.names[: len(keys)]]))
-            if key is not None
-            else (lambda r: (0,))
-        )
+        def key_fn(r):
+            if sort_key is not None:
```

Comment:
nit: is this slightly more accurate since we are iterating over `keys`?
```suggestion
            if keys:
```

Reply:
you got me there

Reply:
updated, thanks!